packages/client/test/routings.spec.ts

```diff
@@ -0,0 +1,269 @@
+import { type PeerRouting, peerRouting } from '@libp2p/interface/peer-routing'
+import { createEd25519PeerId } from '@libp2p/peer-id-factory'
+import { expect } from 'aegir/chai'
+import { create as createIpnsRecord, marshal as marshalIpnsRecord } from 'ipns'
+import all from 'it-all'
+import drain from 'it-drain'
+import { CID } from 'multiformats/cid'
+import { concat as uint8ArrayConcat } from 'uint8arrays/concat'
+import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
+import { createDelegatedRoutingV1HttpApiClient, type DelegatedRoutingV1HttpApiClient } from '../src/index.js'
+
+if (process.env.ECHO_SERVER == null) {
+  throw new Error('Echo server not configured correctly')
+}
+
+const serverUrl = process.env.ECHO_SERVER
+
+describe('libp2p content-routing', () => {
+  let client: DelegatedRoutingV1HttpApiClient
+
+  beforeEach(() => {
+    client = createDelegatedRoutingV1HttpApiClient(new URL(serverUrl))
+  })
+
+  afterEach(async () => {
+    if (client != null) {
+      client.stop()
+    }
+
+    const res = await fetch(`${process.env.ECHO_SERVER}/reset-call-count`)
+    await res.text()
+  })
+
+  it('should provide a content routing implementation', () => {
+    const routing = getContentRouting(client)
+
+    expect(routing).to.be.ok()
+  })
+
+  it('should find providers', async () => {
+    const routing = getContentRouting(client)
+
+    if (routing == null) {
+      throw new Error('ContentRouting not found')
+    }
+
+    const providers = [{
+      Protocol: 'transport-bitswap',
+      Schema: 'bitswap',
+      Metadata: 'gBI=',
+      ID: (await createEd25519PeerId()).toString(),
+      Addrs: ['/ip4/41.41.41.41/tcp/1234']
+    }, {
+      Protocols: ['transport-bitswap'],
+      Schema: 'peer',
+      Metadata: 'gBI=',
+      ID: (await createEd25519PeerId()).toString(),
+      Addrs: ['/ip4/42.42.42.42/tcp/1234']
+    }, {
+      ID: (await createEd25519PeerId()).toString(),
+      Addrs: ['/ip4/43.43.43.43/tcp/1234']
+    }]
+
+    const cid = CID.parse('QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn')
+
+    // load providers for the router to fetch
+    await fetch(`${process.env.ECHO_SERVER}/add-providers/${cid.toString()}`, {
+      method: 'POST',
+      body: providers.map(prov => JSON.stringify(prov)).join('\n')
+    })
+
+    const provs = await all(routing.findProviders(cid))
+    expect(provs.map(prov => ({
+      id: prov.id.toString(),
+      multiaddrs: prov.multiaddrs.map(ma => ma.toString())
+    }))).to.deep.equal(providers.map(prov => ({
+      id: prov.ID,
+      multiaddrs: prov.Addrs
+    })))
+  })
+
+  it('should provide without error', async () => {
+    const routing = getContentRouting(client)
+
+    if (routing == null) {
+      throw new Error('ContentRouting not found')
+    }
+
+    const cid = CID.parse('QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn')
+
+    await routing.provide(cid)
```

Comment:
missing assertion?

```suggestion
    expect(async () => await routing.provide(cid)).not.toThrow()
```

Reply:
I've added an assertion.

Also see the `chai-as-promised` part of https://github.com/ipfs/helia-delegated-routing-v1-http-api/pull/44#discussion_r1376015208